### PR TITLE
Logger add possibility to make logs output more compact

### PR DESCRIPTION
**Pull request by ktrzcinx**
Add option to set timestamp precision and second one to cutout trace location

Original output:
```
$ ./sof-logger -l sof-cnl.ldc -i dma_trace_1.bin
         TIMESTAMP              DELTA C# COMPONENT          LOCATION                      CONTENT
[   497588.489583] (   497588.500000) c0 DMA                .../intel/cavs/hda-dma.c:407  hda-dmac: 4 channel 6 -> get
[   497816.250000] (      227.760422) c0 DMA                .../intel/cavs/hda-dma.c:589  hda-dmac: 4 channel 6 -> config
[   497899.270833] (       83.020836) c0 DMA                .../intel/cavs/hda-dma.c:464  hda-dmac: 4 channel 6 -> start
[   497913.906250] (       14.635417) c0 DMA                .../intel/cavs/hda-dma.c:329  hda-dmac: 4 channel 6 -> enable
[   498012.083333] (       98.177086) c0 SA                          src/lib/agent.c:56   perf sys_load peak plat 25185 cpu 73998
[   498029.322917] (       17.239584) c0 SCHEDULE_LL        ./schedule/ll_schedule.c:178  perf ll_work peak plat 468 cpu 2266
[   498062.968750] (       33.645832) c0 SCHEDULE_LL        ./schedule/ll_schedule.c:304  task add 0x9e0c0080 dma-trace-task <2b972272-c5b1-4b7e-926f-0fc5cb4c4690>
```

With flag -L:
```
$ ../../../build/logger/sof-logger -l sof-cnl.ldc -i dma_trace_1.bin -L
         TIMESTAMP              DELTA C# COMPONENT          CONTENT
[   497588.489583] (   497588.500000) c0 DMA                hda-dmac: 4 channel 6 -> get
[   497816.250000] (      227.760422) c0 DMA                hda-dmac: 4 channel 6 -> config
[   497899.270833] (       83.020836) c0 DMA                hda-dmac: 4 channel 6 -> start
[   497913.906250] (       14.635417) c0 DMA                hda-dmac: 4 channel 6 -> enable
[   498012.083333] (       98.177086) c0 SA                 perf sys_load peak plat 25185 cpu 73998
[   498029.322917] (       17.239584) c0 SCHEDULE_LL        perf ll_work peak plat 468 cpu 2266
[   498062.968750] (       33.645832) c0 SCHEDULE_LL        task add 0x9e0c0080 dma-trace-task <2b972272-c5b1-4b7e-926f-0fc5cb4c4690>
```

With flag -L -f0:
```
$ ../../../build/logger/sof-logger -l sof-cnl.ldc -i dma_trace_1.bin -L -f0
   TIMESTAMP        DELTA C# COMPONENT          CONTENT
[    497588] (    497588) c0 DMA                hda-dmac: 4 channel 6 -> get
[    497816] (       228) c0 DMA                hda-dmac: 4 channel 6 -> config
[    497899] (        83) c0 DMA                hda-dmac: 4 channel 6 -> start
[    497914] (        15) c0 DMA                hda-dmac: 4 channel 6 -> enable
[    498012] (        98) c0 SA                 perf sys_load peak plat 25185 cpu 73998
[    498029] (        17) c0 SCHEDULE_LL        perf ll_work peak plat 468 cpu 2266
[    498063] (        34) c0 SCHEDULE_LL        task add 0x9e0c0080 dma-trace-task <2b972272-c5b1-4b7e-926f-0fc5cb4c4690>
[    498077] (        14) c0 SCHEDULE_LL        task params pri 4 flags 0 start 500000 period 500000
[    498093] (        16) c0 SCHEDULE_LL        num_tasks 2 total_num_tasks 2
[    532524] (     34431) c0 IPC                ipc: hdr 0x30010000 tx (76) > rx (28)
[    532537] (        13) c0 IPC                ipc: pipe 1 comp 0 -> new (type 1)
[    532568] (        31) c0 COMP               comp new host <8b9d100c-6d78-418f-90a3-e0e805d0852b> type 1 pipe_id 1 id 0
```

### CODE REVIEW

**Comment by ktrzcinx** [comment]
@lgirdwood updated

**Comment by lgirdwood** [comment]
@ktrzcinx the updates look good, but I mean paste the before/after into the commit message not the PR message. This way it's obvious for anyone reading the git logs as the current messages are vague.

**Comment by ktrzcinx** [comment]
@lgirdwood updated

File: tools/logger/convert.c [hunk 1]
```diff
@@ -204,13 +204,18 @@ static double to_usecs(uint64_t time, double clk)
 }
```
Comment (dbaluta) [inline tools/logger/convert.c:204]: @ktrzcinx no need for '```' markers in the commit message.

Comment (ktrzcinx) [reply tools/logger/convert.c:204]: updated

**Comment by lgirdwood** [comment]
@ktrzcinx can you double check CI, it looks like the build has failed ? Seems unrelated to this ?

**Comment by ktrzcinx** [comment]
```
Exception ValueError occurred at 'signal.py':47
                        signal only works in main thread
```

and
```
May 10 22:34:30 jf-apl-up2-nocodec-1 kernel: [  970.716052] sof-audio-pci 0000:00:0e.0: error: ipc timed out for 0x80010000 size 216
```

So nothing related with logger, especially when I added functionality is disabled in CI tests (need to pass extra flags)

File: tools/logger/convert.c [hunk 2]
```diff
@@ -346,14 +359,20 @@ static void print_entry_params(FILE *out_fd,
...
+			fprintf(out_fd, "(%s:%u) ",
+				format_file_name(entry->file_name, raw_output),
+				entry->header.line_idx);
 	} else {
 		/* timestamp */
-		fprintf(out_fd, "%s[%16.6f] (%16.6f)%s ",
+		snprintf(time_fmt, sizeof(time_fmt),
+			 "%%s[%%%d.%df] (%%%d.%df)%%s ",
+			 float_precision + 10, float_precision,
+			 float_precision + 10, float_precision);
```
Comment (marc-hb) [inline tools/logger/convert.c:374]: With a crazy long precision `snprintf` can truncate `time_fmt`. Reported by gcc 9.3 -Werr  and fixed in PR #3411

**Pull request merged**